aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -0,0 +1,541 @@
+* The Kestrel transport layer has been exposed as a public interface in `Connections.Abstractions`.
+* Ambiguity between headers and trailers has been resolved by moving trailing headers to a new collection.
+* Synchronous IO APIs, such as `HttpReqeuest.Body.Read`, are a common source of thread starvation leading to app crashes. In 3.0, `AllowSynchronousIO` is disabled by default.
+
+For more information, see <xref:migration/22-to-30>.
+
+## Hosting EventSourceProvider new request counters
+
+The Hosting EventSource (Microsoft.AspNetCore.Hosting) emits the following EventCounters related to incoming requests:
+
+* `requests-per-second`
+* `total-requests`
+* `current-requests`
+* `failed-requests`
+
+## Endpoint routing
+
+See https://github.com/aspnet/AspNetCore.Docs/issues/14291
+
+## Health Checks
+
+Health Checks use endpoint routing with the Generic Host. In `Startup.Configure`, call `MapHealthChecks` on the endpoint builder with the endpoint URL or relative path:
+
+```csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health");
+});
+```
+
+Health Checks endpoints can:
+
+* Specify one or more permitted hosts/ports.
+* Require authorization.
+* Require CORS.
+
+For more information, see the following articles:
+
+* <xref:migration/22-to-30#health-checks> <!-- ignore warning per @guardrex -->
+* <xref:host-and-deploy/health-checks>
+
+## Pipes on HttpContext
+
+It is now possible to read the request body and write the response body using the <xref:System.IO.Pipelines> API. The <!-- <xref:Microsoft.AspNetCore.Http.HttpRequest.BodyReader> --> `HttpRequest.BodyReader` property provides a <xref:System.IO.Pipelines.PipeReader> that can be used to read the request body. The <!-- <xref:Microsoft.AspNetCore.Http.> --> `HttpResponse.BodyWriter` property provides a <xref:System.IO.Pipelines.PipeWriter> that can be used to write the response body.
+
+<!-- indirectly related, https://github.com/dotnet/docs/pull/14414 won't be published by 9/23  -->
+
+## Worker service and SDK
+
+See https://github.com/aspnet/AspNetCore.Docs/issues/14269
+
+## Forwarded Headers Middleware improvements
+
+In previous versions, calling <xref:Microsoft.AspNetCore.Builder.HstsBuilderExtensions.UseHsts*> and  <xref:Microsoft.AspNetCore.Builder.HttpsPolicyBuilderExtensions.UseHttpsRedirection*> were problematic when deployed to an Azure Linux or behind any reverse proxy other than IIS. The fix for previous versions is documented in [Forward the scheme for Linux and non-IIS reverse proxies](xref:host-and-deploy/proxy-load-balancer#forward-the-scheme-for-linux-and-non-iis-reverse-proxies).
+
+In ASP.NET Core 3.0, this problem has been fixed. The host enables the [Forwarded Headers Middleware](xref:host-and-deploy/proxy-load-balancer#forwarded-headers-middleware-options) when the `ASPNETCORE_FORWARDEDHEADERS_ENABLED` environment variable has been set to `true`.
+
+## HTTP/2 enabled by default
```

Comment:
I'm probably a bit biased, but I would love this to be a little closer to the top ;). I think it's a fairly important feature and it brings us up to parity with most of our competitive frameworks/platforms.

Reply:
And put it next to/near the Kestrel section, as it only applies to Kestrel.